coordinator/src/test/scala/filodb.coordinator/queryplanner/SingleClusterPlannerSpec.scala

```diff
@@ -1312,4 +1313,104 @@ class SingleClusterPlannerSpec extends AnyFunSpec with Matchers with ScalaFuture
       }
     }
   }
+
+  it("should propagate new filters to FunctionArgs") {
+    // not intended to be all-encompassing; just specific to this test
+    def getLeafFilterGroups(plan: LogicalPlan): Seq[Seq[ColumnFilter]] = {
+      val res = plan match {
+        case sqw: SubqueryWithWindowing =>
+          sqw.functionArgs.flatMap(getLeafFilterGroups(_)) ++ getLeafFilterGroups(sqw.innerPeriodicSeries)
+        case psw: PeriodicSeriesWithWindowing =>
+          psw.functionArgs.flatMap(getLeafFilterGroups(_)) ++ getLeafFilterGroups(psw.series)
+        case aif: ApplyInstantFunction =>
+          aif.functionArgs.flatMap(getLeafFilterGroups(_)) ++ getLeafFilterGroups(aif.vectors)
+        case air: ApplyInstantFunctionRaw =>
+          air.functionArgs.flatMap(getLeafFilterGroups(_)) ++ getLeafFilterGroups(air.vectors)
+        case svd: ScalarVaryingDoublePlan =>
+          svd.functionArgs.flatMap(getLeafFilterGroups(_)) ++ getLeafFilterGroups(svd.vectors)
+        case nlp: NonLeafLogicalPlan =>
+          nlp.children.flatMap(getLeafFilterGroups(_))
+        case rs: RawSeries =>
+          Seq(rs.filters)
+        case _ => throw new IllegalArgumentException(s"unhandled type: ${plan.getClass}")
+      }
+      res.filter(_.nonEmpty)
+    }
+
+    // the filters for all plans that use them
+    val initialFilters = Seq(
+      ColumnFilter("foo", Equals("bar")),
+      ColumnFilter("lname1", Equals("lvalA"))  // should be replaced by updFilters
+    )
+    // the filters we'll pass to replaceFilters()
+    val updFilters = Seq(
+      ColumnFilter("lname1", Equals("lval1")),
+      ColumnFilter("lname2", Equals("lval2"))
+    )
+    // the filters we expect to see post-replaceFilters() in any plan that has them
+    val expFilters = Seq(
+      ColumnFilter("foo", Equals("bar")),
+      ColumnFilter("lname1", Equals("lval1")),
+      ColumnFilter("lname2", Equals("lval2"))
+    )
+
+    // dummy plans
+    val rawSeries = RawSeries(
+      IntervalSelector(1, 100),
+      initialFilters,
+      Nil
+    )
+    val periodicSeries = PeriodicSeries(
+      rawSeries,
+      1, 10, 100
+    )
+    val funcArg = ScalarVaryingDoublePlan(periodicSeries, ScalarFunctionId.Scalar, Nil)
+
+    val plans = Seq(
```

Comment:
These are manually constructed because:
- the parser forbids `-WithWindowing` types with non-fixed args, so we can't otherwise test whether-or-not `scalar()`-wrapped vector args are updated by `replaceFilters`. (99% sure something like `quantile_over_time(scalar(foo{}), bar{}[])` is valid PromQL).
- `ApplyInstantFunctionRaw` and `ScalarVaryingDoublePlan`s are never constructed with `ColumnFilter`-containing args when these plans are parsed out of a query string, so we wouldn't be able to test `replaceFilters` for them, either.